### ISSUE TITLE
Articullate api is responding very slow

### ISSUE THREAD

**Issue by ferozewaris** [samtecspg/articulate#569]
### Issues

Hi I am using articulate api version 0.0.1 along with rasa NLU version 0.13.0. Whenever i call agent converse api i get very slow response i also did the stress testing by sending 25 concurrent requests results of that test is attached
![screen shot 2019-02-06 at 7 47 48 pm](https://user-images.githubusercontent.com/22313397/52574101-56767f80-2e3d-11e9-8c99-d6088a5ee70e.png)

You can clearly see from the above image that the mean response time is about 25 seconds. 
I am using CentOS machine with 4 cores and 8 gb memory other services are also running on that machine but they are taking very less whenever i start the stress test the CPU usage of articulate shoots upto about 300% and average load reaches upto 4.

What could be the problem here ?  

Do let me know if any other information is required

----

**Comment by wrathagom** [comment]
Can you tell me how many domains you have? can we get a screenshot/video of `docker stats` while the test is taking place?

**Comment by wrathagom** [comment]
Same question asked on SO: https://stackoverflow.com/questions/54634557/spg-articulate-api-is-responding-very-slow

**Comment by ferozewaris** [comment]
Hi sorry for the late reply actually i couldnt be able to test again on the same machine so i setup another machine with 8 cores and 16 GB RAM and tested on it the cpu usage reduced to about 100 to 150 % during test but the response time remained approximately same . There is only one domain in articulate. screenshots of docker stats during test and test results are attached kindly see
![screen shot 2019-02-13 at 4 29 43 pm](https://user-images.githubusercontent.com/22313397/52710180-7e3a2480-2fb0-11e9-99fd-049ee7fccd18.png)
![screen shot 2019-02-13 at 4 51 57 pm](https://user-images.githubusercontent.com/22313397/52710193-898d5000-2fb0-11e9-83e0-a51559a3c454.png)

**Comment by wrathagom** [comment]
Can you describe your network/Docker setup? I notice you're connecting using https, but to a local IP?

I am working up some info on this issue today.

**Comment by ferozewaris** [comment]
yes the api is listening on https with self signed certificates and this machine is on LAN.

**Comment by wrathagom** [comment]
I just noticed that your docker stats doesn't include the full stack, can you post a screenshot showing everything. The API CPU usage actually isn't what I want to see, I want to see what Rasa is doing.

**Comment by ferozewaris** [comment]
Hi I have again tested it and recorded stats of Rasa, api, duckling and Redis. Please see the attached video. Kindly download the zip file and extract the video. Also results of stress test are also attached
![screen shot 2019-02-18 at 11 56 20 am](https://user-images.githubusercontent.com/22313397/52933186-40ece280-3374-11e9-8956-dae7412bf244.png)

[stress-test recording.mov.zip](https://github.com/samtecspg/articulate/files/2874265/stress-test.recording.mov.zip)
